modules/javafx.graphics/src/main/java/com/sun/javafx/sg/prism/NGPointLight.java

```diff
@@ -26,11 +26,85 @@
 package com.sun.javafx.sg.prism;
 
 /**
- * TODO: 3D - Need documentation
+ * The peer of the {@code PointLight} class. Holds the default values of {@code PointLight}'s
+ * properties and updates the visuals via {@link NGNode#visualsChanged} when one of the current
+ * values changes. The peer receives its changes by {@code PointLight.doUpdatePeer} calls.
  */
 public class NGPointLight extends NGLightBase {
 
+    /** Constant attenuation factor default value */
+    private static final double DEFAULT_CA = 1;
+    /** Linear attenuation factor default value */
+    private static final double DEFAULT_LA = 0;
+    /** Quadratic attenuation factor default value */
+    private static final double DEFAULT_QA = 0;
+    /** Max range default value */
+    private static final double DEFAULT_MAX_RANGE = Double.POSITIVE_INFINITY;
+
     public NGPointLight() {
     }
 
-}
+    public static double getDefaultCa() {
+        return DEFAULT_CA;
+    }
+
+    public static double getDefaultLa() {
+        return DEFAULT_LA;
+    }
+
+    public static double getDefaultQa() {
+        return DEFAULT_QA;
+    }
+
+    public static double getDefaultMaxRange() {
+        return DEFAULT_MAX_RANGE;
+    }
+
+
+    private double ca = DEFAULT_CA;
```

Comment:
The coefficients are not directly used in any arithmetic on java side. They are converted to `float` and passed to GL or D3D pipeline. Should these be `float `instead of `double` ?

Reply:
I was wondering about it myself, but all the other values are `double`s that are cast to `float`s. Wouldn't it also be odd to have the API properties `DoubleProperty` and the peer to use `float`s? Isn't it just a matter of where the cast happens?

Reply:
Yes, it's just a matter of where the cast happens. While we aren't entirely consistent, I think you'll find more places where the peers hold floats rather than double (one notable exception is transforms where we need the extra precision). In particular, Color and position values for shapes are stored in the peer as floats. Also, PhongMaterial stores the power as a float even though the API is double.

Reply:
Alright, I don't mind it, I'll cast in the peer level.